compute/aws-ec2.js

```diff
@@ -104,6 +104,54 @@ class EC2 {
     });
   }
 
+  /**
+   * Start instance monitoring
+   * @monitor
+   * @param {object} params
+   */
+  monitor(params) {
+    return new Promise((resolve, reject) => {
+      this._ec2.monitorInstances(params, function(err, data) {
+        if (err && err.code === 'DryRunOperation') {
+          params.DryRun = false;
+          this._ec2.monitorInstances(params, function(err, data) {
+              if (err) {
+                reject(err);
+              } else if (data) {
+                resolve(data);
+              }
+          });
+        } else {
+          reject('Permission denied');
```

Comment:
Are we exactly sure `err` would be only for permission issues?

Reply:
I will correct that :) . According to the AWS documentation the required permissions are there, the request returns `DryRunOperation`; otherwise, it returns `UnauthorizedOperation`. I will add another if condition here.

Related link - https://docs.aws.amazon.com/AWSEC2/latest/APIReference/CommonParameters.html